### ISSUE TITLE
Improve Image insertion widget to support better titling of images. 

### ISSUE THREAD

**Issue by iandstanley** [decaporg/decap-cms#6761]
**Is your feature request related to a problem? Please describe.**

I am new to decap cms and have recently used it to write a website for a local charity. One question that came up on handover was how to title an inserted picture. The default image widget does not appear to support a title in the way a figure can be used. Yes images have titles but these are mapped to roll over but the average Jane Public knows nothing about this.

A workaround I have suggested is to repeat the title line below the image widget, but this confuses the end user.

**Describe the solution you'd like**

Ideally: 

As the alt-text is used for SEO,  screen readers and missing image loads, the Title field is only used for rollover titles which is little use for most websites.

I would like to see the image widget either use a figure if a title is inserted or have an option to use title as a caption and the 

Having a single image widget makes it easier for novice end users who expect a title field to subtitle the image inserted. This of course will complicate round trip processing, so the alternative below may be desired instead.

Alternatively:

Have an  insert figure widget with a figure caption that is mapped to a field in the widget.

**Describe alternatives you've considered**

Asking end users to add a title below the image causes two issues:
1. End users just get confused with adding a title to the image AFTER the widget when there is a perfectly good field in the widget.
2. Having the website styled correctly becomes an issue. Making sure the end user actually uses a specific formatting (eg. a <h2> for the title below the image) is almost impossible over even a short period of time as end users don't remember from month to month that they are supposed to use a H2 tag as per the designers instructions. The result is that the website ends up with a random selection of <b>, <h1>, <h2> and other tags which may each have a different styling in the overall design.

**Additional context**

**Comment by martinjagodic** [comment]
@iandstanley did you issue #857? I assume that you want what is discussed there.

**Comment by nternetinspired** [comment]
The image widget allows you to add an image to a markdown file. Obviously, it has to output Markdown and Markdown does not support figure elements.

The image widget does exactly what is required, adding a title element to an image, per Markdown spec. Changing that would create invalid Markdown.

I saw your post on the Hugo forum. The good news is that in Hugo you can use an image render hook to parse the image reference in Markdown and output a figure element, if you wish. Alternatively, enable unsafeHhtml in Hugo and you can add whatever html you want to your Markdown files.

TL:DR, Figures with captions are trivial to implement in Hugo.

**Comment by nternetinspired** [comment]
> Yes images have titles but these are mapped to roll over

That is how the image element works. It has nothing to do with Decap.

**Comment by martinjagodic** [comment]
Closing as stale
